packages/react/toggle-button/src/ToggleButton.stories.tsx

```diff
@@ -16,6 +17,154 @@ export const Controlled = () => {
+          console.log('Button 3 has changed!');
+        }}
+      >
+        Option 3
+      </ToggleButton>
+    </ToggleButtonGroup>
+  );
+};
+
+export const GroupedWithRovingFocus = () => {
+  return (
+    <div>
+      <VisuallyHidden id="desc">
+        To navigate the buttons within the group, use the arrow keys
+      </VisuallyHidden>
+      <ToggleButtonGroup
+        aria-label="Options"
+        aria-describedby="desc"
+        defaultValue={['1']}
+        rovingFocus
+      >
+        <ToggleButton value="1" as={StyledRoot}>
+          Option 1
+        </ToggleButton>
+        <ToggleButton value="2" as={StyledRoot}>
+          Option 2
+        </ToggleButton>
+        <ToggleButton
+          value="3"
+          as={StyledRoot}
+          onToggledChange={() => {
+            // Make sure onToggledChange fires even in grouped buttons
+            console.log('Button 3 has changed!');
+          }}
+        >
+          Option 3
+        </ToggleButton>
+      </ToggleButtonGroup>
+    </div>
+  );
+};
+
+export const GroupedControlled = () => {
+  const [selectionIsRequired, setSelectionIsRequired] = React.useState(false);
+  const [value, setValue] = React.useState<string[]>(['1']);
+
+  return (
+    <div>
+      <div>
+        <label>
+          <input
+            type="checkbox"
+            checked={selectionIsRequired}
+            onChange={(e) => setSelectionIsRequired(e.target.checked)}
+          />
+          <span>Ensure selection</span>
+        </label>
+      </div>
+      <hr />
+      <div>
+        <ToggleButtonGroup
+          aria-label="Options"
+          value={selectionIsRequired && (value == null || value.length < 1) ? ['1'] : value}
+          onValueChange={setValue}
+          selectionIsRequired={selectionIsRequired}
+        >
+          <ToggleButton value="1" as={StyledRoot}>
+            Option 1
+          </ToggleButton>
+          <ToggleButton value="2" as={StyledRoot}>
+            Option 2
+          </ToggleButton>
+          <ToggleButton
+            value="3"
+            as={StyledRoot}
+            onToggledChange={() => {
+              // Make sure onToggledChange fires even in grouped buttons
+              console.log('Button 3 has changed!');
+            }}
+          >
+            Option 3
+          </ToggleButton>
+        </ToggleButtonGroup>
+      </div>
+    </div>
+  );
+};
+
+export const GroupedControlledExclusive = () => {
+  const [selectionIsRequired, setSelectionIsRequired] = React.useState(false);
+  const [value, setValue] = React.useState<string | null>(null);
+
+  return (
+    <div>
+      <div>
+        <label>
+          <input
+            type="checkbox"
+            checked={selectionIsRequired}
+            onChange={(e) => setSelectionIsRequired(e.target.checked)}
+          />
+          <span>Ensure selection</span>
+        </label>
+      </div>
+      <hr />
+      <div>
+        <ToggleButtonGroupExclusive
```

Comment:
@benoitgrelard and I have gone over this a few times in Slack and I think we both feel like having two components to handle single vs. multiple values is a good solution, though I don't think either of us think the naming is currently the best. I'll try not to speak for him, so B feel free to weigh in with whatever thoughts you have on this point.

> Also, I wasn't really party to the conversations regarding merging `ToggleButton` and `ToggleButtonGroup` but the reason they were merged in `RadioGroup` was because no one would use `Radio` on its own (it's not exported from `index.ts`). Are we saying here that no one would use `ToggleButton` on its own now?

That is my understanding as well. `ToggleButton` can absolutely be used on its own.

That said, re-exporting it from a separate `ToggleGroup` component was something I considered as well. This was another point we discussed outside of GitHub and figured most folks will probably reach for the group for most cases somewhere in their app, but TBH with the new Parcel build being so freaking fast I don't know that it hurts to create an extra component after all. 

All that 

> As far as multi vs single, we managed to keep the API consistent with `Slider` for this to avoid a separate component (always accepts array type) but I agree it doesn't look like we can do that here. If there is a way to get types working I'd prefer not to push a new component onto consumer personally but if not, fine. If we were unable to do this for `Slider` I'd have created a separate component too.

I was kind of hoping you might have some wizardry ideas here 😅 As far as I can tell, `forwardRefWithAs` limits us a bit in terms of making generic components with other dynamically typed props because we cast the inner function. We could potentially bypass it but I didn't spend much time on it here because I wanted to avoid the time sink if possible. I do think this is a bit different than `Slider` though since changing the top-level value is adding/removing items from the array vs. changing values within it. `MultiSelectToggleGroup` is only weird to me because I feel like multiple selection should be the default behavior. I don't hate `SingleSelectToggleGroup` though. @benoitgrelard any thoughts to add here?

Reply:
> but TBH with the new Parcel build being so freaking fast I don't know that it hurts to create an extra component after all.

:+1: Yep, Parcel is on fire now haha. Thanks for that!! Would love it if we could make it separate because that is the approach we've gone with everywhere else.

> I was kind of hoping you might have some wizardry ideas here 😅

I've actually been thinking about this haha. I'm gonna play some more today as I was getting somewhere in a CodeSandbox yesterday (it worked) but the same code wasn't behaving the same in our package ☹️ Need to try figure out wtf is going on there.

> `MultiSelectToggleGroup` is only weird to me because I feel like multiple selection should be the default behaviour

On the surface, `MultiSelectToggleGroup` seems to be just a bunch of `ToggleButton`s that the user could control themselves. I initially wondered why we need a component for that at all but I then realised there is a bit more to it than that. The single version though, much like `RadioGroup` handles more obvious additional complexity so is more likely to be something someone would need to reach to a component for. That's why I felt the single version should be the default. Plus, it keeps things consistent with `RadioGroup` which is an added bonus haha. 

Agree we probably need @benoitgrelard and @colmtuite's thoughts on this one. Which should be the default guyzzzzz?

Reply:
Had a chat with @colmtuite about this yesterday, I'd be good either way really, although for me the naming then would become quite important and I don't love what we have so could be swayed.

I think @colmtuite was thinking the same as you @jjenzz, that users are more likely to look for the single one.
But Colm can confirm, perhaps I misunderstood.

Reply:
I have personally used multiple select toggle buttons in groups more often than not, but I'm fine if we want to make single the default and create a `MultiSelectToggleGroup` if that feels better for everyone else.

> On the surface, `MultiSelectToggleGroup` seems to be just a bunch of `ToggleButton`s that the user could control themselves. I initially wondered why we need a component for that at all but I then realised there is a bit more to it than that.

Yeah, I thought about that as well and we discussed just having examples to show grouping and not having a package for that. In the simplest case you are right, but it's nice if we can abstract the aria and also provide roving focus support in the cases where it's needed.

Sounds like we might want @colmtuite to weigh in, but if I'm reading this right it sounds like we'd all be a little happier with something like this:
- `toggle-button`: This package is unchanged
- `toggle-group`: This package composes `toggle-button` and assumes users select a single `toggle-button` in a group similar to a `radio` in a `radio-group`
- `multi-select-toggle-group`: This package composes `toggle-button` and assumes users can select any number of `toggle-button`s in a group similar to a `checkbox` in a `checkbox-group`

I think if we go this route we can also probably axe the `required` prop altogether. If a consumer wants to require a selection they can do so by controlling state, and they can implement their own `required` prop if they want by wrapping the primitive. This seems better to me since these aren't generally going to be form fields, and I associate `required` with inputs that would be validated in a form context. Any thoughts here one way or the other?

Reply:
Yep that's in line with what I was saying on our call @benoitgrelard. I think it's a very close call, both are used a lot, from what I've seen on the web. But I'd vote that the single select variant is used more commonly. For example, it's all over Modulz, Figma, and Sketch and we never use the multi select variant. Google Material (not talking about MaterialUI) also default to the single select variant.

I'd still prefer a single package with a prop, but something something type issues.

So yeah, I'd be happy with that proposed set-up Chance just posted.

Reply:
Yeah in an ideal the group one would be one package with a prop. It "may" be possible still, we were close with Jenna yesterday…

Reply:
@benoitgrelard In that case perhaps I'll stick with one package for `toggle-group` for now. We can still export both as separate components in the mean time, but perhaps simpler to deprecate the component later if can fix the types vs. deprecating an entire package.

Reply:
Are we going with ToggleGroup (toggle-group) or ToggleButtonGroup? (toggle-button-group)?

I'm leaning towards the latter as the base component is `ToggleButton` and base package is `toggle-button`.

Otherwise with the other I'd think the base one would have to be `Toggle` in a `toggle` package.

Thoughts?

Reply:
I'm OK with either. `MultiSelectToggleButtonGroup` is a bit of a mouthful though, so removing `Button` makes that one much better IMO.

Reply:
> Are we going with `ToggleGroup` (`toggle-group`) or `ToggleButtonGroup`? (`toggle-button-group`)?

I don't think that just because it is wrapping `ToggleButton` means it should be called  `ToggleButtonGroup` personally. There is no need to expose that implementation detail in the new component. Otherwise, that opens questions like, should `Accordion` be called `MultiSelectCollapsible`?

Naming should always describe what the new thing is and not what it uses under the hood IMO. Yes, `ToggleButtonGroup` also fits that requirement but so does `ToggleGroup` & `ToggleGroupItem` and they're shorter and follow the `RadioGroup` & `RadioGroupItem` pattern. Or do you think we should rename `RadioGroup` to `RadioInputGroup`?

Reply:
> but it's nice if we can abstract the aria and also provide roving focus support in the cases where it's needed.

Agreed 🙂 

> - `toggle-button`: This package is unchanged
> - `toggle-group`: This package composes `toggle-button` and assumes users select a single `toggle-button` in a group similar to a `radio` in a `radio-group`
> - `multi-select-toggle-group`: This package composes `toggle-button` and assumes users can select any number of `toggle-button`s in a group similar to a `checkbox` in a `checkbox-group`

Yep, this is exactly what I posted in my original comment so gets my vote 💯 :+1: 🎉

Reply:
Ok, I've reverted everything in `toggle-button` and moved all new code into a new `toggle-group` package. I'm going to expose `ToggleGroup` and `MultiSelectToggleGroup` from that package for now, with the goal of eventually having a generic typed `ToggleGroup` component that handles selection mode via a prop. If that turns out to be more trouble than it's worth we can always create a new `multi-select-toggle-group` package later if we'd prefer.